ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java

```diff
@@ -197,10 +194,11 @@ TopoAutomatonSimulationResult simulateTopologicalAutomatons(State automatonState
         // -- First get forced network actions
         Set<NetworkAction> appliedNetworkActions = crac.getNetworkActions(automatonState, UsageMethod.FORCED);
 
-        // -- Then add those with an OnFlowConstraint usage rule if their constraint is verified
+        // -- Then add those with TO_BE_EVALUATED usage method when evaluation condition is verified
+        // -- Evaluation condition is isAnyMarginNegative amongst network actions' flow cnecs associated to their usage rules
         crac.getNetworkActions(automatonState, UsageMethod.TO_BE_EVALUATED).stream()
-            .filter(na -> RaoUtil.isRemedialActionAvailable(na, automatonState, prePerimeterSensitivityOutput, crac.getFlowCnecs(), network, raoParameters.getObjectiveFunctionParameters().getType().getUnit()))
-            .forEach(appliedNetworkActions::add);
+                .filter(na -> na.isRemedialActionAvailable(automatonState, RaoUtil.isAnyMarginNegative(prePerimeterSensitivityOutput, na.getFlowCnecsConstrainingUsageRules(crac.getFlowCnecs(), network, automatonState), raoParameters.getObjectiveFunctionParameters().getType().getUnit())))
```

Comment:
I'm not a big fan of calling a method on the remedial action (isRemedialActionAvailable) with an argument that's computed by RaoUtil (isAnyMarginNegative) that uses a list of cnecs computed by the remedial action (getFlowCnecsConstrainingUsageRules).
Could isAnyMarginNegative be moved to the remedial action, and made private, as well as the getFlowCnecsConstraingUsageRules method. (Or will there be a circular dependency issue if we try give the sensitivity output to the remedial action?)

Reply:
using a FlowResult in RemedialAction indeed leads to a circular dependency. That's why it's the only method kept in RaoUtil